sentry/src/main/java/io/sentry/SentryIntegrationPackageStorage.java

```diff
@@ -64,12 +67,48 @@ public void addPackage(final @NotNull String name, final @NotNull String version
 
     SentryPackage newPackage = new SentryPackage(name, version);
     packages.add(newPackage);
+    try (final @NotNull ISentryLifecycleToken ignored = mixedVersionsLock.acquire()) {
+      mixedVersionsDetected = null;
+    }
   }
 
   public @NotNull Set<SentryPackage> getPackages() {
     return packages;
   }
 
+  public boolean checkForMixedVersions(final @NotNull ILogger logger) {
+    final @Nullable Boolean mixedVersionsDetectedBefore = mixedVersionsDetected;
+    if (mixedVersionsDetectedBefore != null) {
+      return mixedVersionsDetectedBefore;
+    }
+    try (final @NotNull ISentryLifecycleToken ignored = mixedVersionsLock.acquire()) {
+      final @NotNull String sdkVersion = BuildConfig.VERSION_NAME;
+      boolean mixedVersionsDetectedThisCheck = false;
+
+      for (SentryPackage pkg : packages) {
+        if (pkg.getName().startsWith("maven:io.sentry:")
+            && !sdkVersion.equalsIgnoreCase(pkg.getVersion())) {
+          logger.log(
```

Comment:
I'm thinking if we should introduce a method that force-logs regardless of the debug and diagnosticLevel values set and use it here, wdyt?